src/Tools/dotnet-format/Program.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Invocation;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Tools.CodeFormatter.Logging;
+using Microsoft.CodeAnalysis.Tools.CodeFormatter.MSBuild;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.CodeAnalysis.Tools.CodeFormatter
+{
+    class Program
+    {
+        static async Task<int> Main(string[] args)
+        {
+            var parser = new CommandLineBuilder()
+                .UseParseDirective()
+                .UseHelp()
+                .UseSuggestDirective()
+                .UseParseErrorReporting()
+                .UseExceptionHandler()
+                .AddOption(new[] { "-w", "--workspace" }, "The project or solution to format", a => a.WithDefaultValue(() => null).ExactlyOne())
+                .AddOption(new[] { "-q", "--quiet" }, "Suppresses all output except warnings and errors", a => a.ParseArgumentsAs<bool>((sr) => ArgumentParseResult.Success(true), new ArgumentArityValidator(0, 0)))
+                .AddOption(new[] { "-v", "--verbose" }, "Show verbose output", a => a.ParseArgumentsAs<bool>((sr) => ArgumentParseResult.Success(true), new ArgumentArityValidator(0, 0)))
+                .AddVersionOption()
+                .OnExecute(typeof(Program).GetMethod(nameof(Run)))
+                .Build();
```

Comment:
Parser config looks good cc @jonsequitur

Reply:
That's a good suggestion. =) Thanks @KathleenDollard